lib/_http_outgoing.js

```diff
@@ -109,35 +109,29 @@ OutgoingMessage.prototype.setTimeout = function(msecs, callback) {
 // any messages, before ever calling this.  In that case, just skip
 // it, since something else is destroying this connection anyway.
 OutgoingMessage.prototype.destroy = function(error) {
-  if (this.socket)
+  if (this.socket) {
+    // If the socket is corked from a write, uncork it before destroying it.
+    if (this.socket._writableState.corked)
+      this.socket.uncork();
```

Comment:
AFAIK that would be a backwards-incompatible change. It would change behavior in something like this:
```javascript
(req, res) => {
  res.cork();
  res.write('should not write');
  res.destroy();
}
```
Because `Socket#destroy()` stops all IO on the socket, it isn't supposed to flush the last message. With this commit, writing to the stream will only cork it once, and thus we only uncork once, to leave the dev's previous cork-intent there. Looping until we uncork fully would write the message which I don't believe happens right now (edge case, yes 😛 )